cityofaustin/atd-moped
# Manage dynamic interactive layer IDs and layers (#5)

## Associated issues
https://github.com/cityofaustin/atd-data-tech/issues/10407

Changes:
- Updates from conditionally rendering Mapbox sources and layers with state conditionals to conditionally toggling [layer visibility ](https://docs.mapbox.com/mapbox-gl-js/style-spec/layers/#layout-background-visibility)
   - https://docs.mapbox.com/mapbox-gl-js/example/toggle-layers/
- Adds `isInteractive` boolean to the layers config so we can generate a list of [interactive layer IDs](https://visgl.github.io/react-map-gl/docs/api-reference/map#interactivelayerids) from it

## Testing
**URL to test:** 

https://md-10407-interactive-layers--atd-moped-main.netlify.app/ or local

**Steps to test:**
1. Go to a new or existing project's **Map** tab
2. Click **New Component** and select the type called **Access Control - Driveway Closure**
3. Click **Continue**, zoom the map in until you see the dashed lines, and select a couple line segments
4. You should see the street labels above the selecting line segments like this example

<img width="439" alt="Screen Shot 2022-10-21 at 1 38 49 PM" src="https://user-images.githubusercontent.com/37249039/197266654-0cf4630c-20f4-4e3b-b8fe-aa9611e40386.png">

5. Change the base map to aerial and you should see the street labels remain above your selected segments

<img width="439" alt="Screen Shot 2022-10-21 at 1 39 28 PM" src="https://user-images.githubusercontent.com/37249039/197266785-6114b3c9-c390-46d5-8b11-6f01f37e937d.png">

6. Test out removing segments by clicking them too and then click **Save**
7. Click on a line segment of your saved component and you should see a pop up appear with details about the component and a link to edit or delete

---
#### Ship list
